src/ghci/parse/lines.rs

```diff
@@ -1,18 +1,25 @@
-use winnow::combinator::terminated;
+use winnow::ascii::line_ending;
+use winnow::ascii::not_line_ending;
+use winnow::combinator::alt;
+use winnow::combinator::eof;
+use winnow::error::ContextError;
+use winnow::error::ErrMode;
 use winnow::stream::AsChar;
+use winnow::stream::Compare;
 use winnow::stream::FindSlice;
+use winnow::stream::SliceLen;
 use winnow::stream::Stream;
 use winnow::stream::StreamIsPartial;
-use winnow::token::take_until0;
 use winnow::PResult;
 use winnow::Parser;
 
 /// Parse the rest of a line, including the newline character.
 pub fn rest_of_line<I>(input: &mut I) -> PResult<<I as Stream>::Slice>
 where
-    I: Stream + StreamIsPartial + for<'i> FindSlice<&'i str>,
+    I: Stream + StreamIsPartial + for<'i> FindSlice<&'i str> + for<'i> Compare<&'i str>,
```

Comment:
What does the `for` keyword do here?

Reply:
It's basically a `forall`. Apparently they're called [higher ranked trait bounds](https://doc.rust-lang.org/nomicon/hrtb.html).